catalog/core/catalog-core-api/src/main/java/ddf/catalog/operation/FacetProperties.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.operation;
+
+import java.io.Serializable;
+import java.util.Set;
+
+/**
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ */
+public interface FacetProperties extends Serializable {
+
+  enum SortFacetsBy {
+    INDEX,
+    COUNT
+  }
+
+  Set<String> getFacetAttributes();
+
+  SortFacetsBy getSortKey();
+
+  int getFacetLimit();
```

Comment:
❓ It looks like this only supports term facets.  Is the idea that we would add new types of facet properties for numeric and date range facet properties?

Reply:
Yes, the plan for now is to iron out what we like and don't like about this faceting API via text facets first, and then add support for date/range type faceting in the future.

Reply:
✏️  You might consider renaming this class to something more specific like `TermFacetProperties` since we plan to add numeric and date facet properties in the future.

Reply:
renamed